kinzhal-processor/src/jvmMain/kotlin/com/daugeldauge/kinzhal/processor/KinzhalSymbolProcessor.kt

```diff
@@ -54,6 +56,53 @@ internal class KinzhalSymbolProcessor(private val codeGenerator: CodeGenerator,
             }
             .toList()
 
+        val assistedFactoryTypes = resolver.getSymbolsWithAnnotation(AssistedFactory::class.requireQualifiedName())
+            .mapNotNull { assistedFactoryDeclaration ->
+                if (assistedFactoryDeclaration !is KSClassDeclaration || !assistedFactoryDeclaration.isAbstract()) {
+                    logger.error("@AssistedFactory can be applied only to abstract types", assistedFactoryDeclaration)
```

Comment:
Could you add tests for all new errors? (see `ProcessorErrorsTest`)

Reply:
Done